pytest_html/plugin.py

```diff
@@ -41,6 +41,9 @@ def pytest_addoption(parser):
     group.addoption('--html', action='store', dest='htmlpath',
                     metavar='path', default=None,
                     help='create html report file at given path.')
+    group.addoption('--self-contained-html', action="store", dest='self_contained_html', default=False,
```

Comment:
You can access it from `config` such as `config.getoption('self_contained_html')`. The `config` is available through many objects such as `request`, `session`, etc.